temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1149,10 +1157,22 @@ private void processFailWorkflowExecution(
           failureType = Optional.of(failureInfo.getType());
           backoffInterval = rs.getBackoffIntervalInSeconds(failureType, store.currentTime());
         }
-      } else {
+      } else if (failure.hasTerminatedFailureInfo()
```

Comment:
We can probably simplify this to
```
failure.hasTerminatedFailureInfo()
          || failure.hasCanceledFailureInfo()
          || failure.hasServerFailureInfo() && failure.getServerFailureInfo().getNonRetryable()
```
since it's only called for workflow tasks.